test/es-module/test-esm-specifiers-symlink.mjs

```diff
@@ -0,0 +1,38 @@
+import * as common from '../common/index.mjs';
+import path from 'path';
+import fs from 'fs/promises';
+import tmpdir from '../common/tmpdir.js';
+import { spawn } from 'child_process';
+import assert from 'assert';
+
+tmpdir.refresh();
+const tmpDir = tmpdir.path;
+
+await fs.mkdir(path.join(tmpDir, 'subfolder/node_modules/package-a'), { recursive: true });
```

Comment:
nit: I think we could move this after the symlink creation in case we end up skipping the test.
```suggestion
```